src/cogent3/app/evo.py

```diff
@@ -104,29 +106,142 @@ def __init__(
+
+        >>> from cogent3 import make_aligned_seqs, get_app
+        >>> aln = make_aligned_seqs({
+        ...    "Human": "ATGCGGCTCGCGGAGGCCGCGCTCGCGGAG",
+        ...    "Mouse": "ATGCCCGGCGCCAAGGCAGCGCTGGCGGAG",
+        ...    "Opossum": "ATGCCAGTGAAAGTGGCGGCGGTGGCTGAG",
+        ... })
+        >>> app = get_app("model", "F81")
+        >>> result = app(aln)
+        >>> print(result.to_rich_dict())
+        {'type': 'cogent3.app.result.model_result', ...
+
+        Call a susbstitution model and fit to an alignment with seqs > 3.
+
+        >>> tree = "(Mouse,(Human,Gorilla),Opossum)"
+        >>> aln2 = make_aligned_seqs({
+        ...      "Human": "ATGCGGCTCGCGGAGGCCGCGCTCGCGGAG",
+        ...      "Gorilla": "ATGCGGCGCGCGGAGGCCGCGCTCGCGGAG",
+        ...      "Mouse": "ATGCCCGGCGCCAAGGCAGCGCTGGCGGAG",
+        ...      "Opossum": "ATGCCAGTGAAAGTGGCGGCGGTGGCTGAG",
+        ... })
+        >>> app_tr = get_app("model", "F81", tree=tree)
+        >>> result = app_tr(aln2)
+        >>> print(result.to_rich_dict())
+        {'type': 'cogent3.app.result.model_result', ...
+
+        Assign a tree function to estimate the phylogenetic tree for the incoming alignment.
+
+        >>> dist_cal = get_app("fast_slow_dist", fast_calc="paralinear", moltype="dna")
+        >>> est_tree = get_app("quick_tree")
+        >>> tree_func = dist_cal + est_tree
+        >>> model = get_app("model", "F81", tree_func=tree_func)
+        >>> result = model(aln2)
+        >>> print(result.to_rich_dict())
+        {'type': 'cogent3.app.result.model_result', ...
+
+        Specify the time-heterogeneous settings for the model fitting.
+        For details, see https://cogent3.org/doc/app/evo-model-timehet#
+
+        >>> app_thet = get_app(
+        ...    "model",
+        ...    "HKY85",
+        ...    tree=tree,
+        ...    time_het=[dict(tip_names=["Human", "Opossum"], outgroup_name="Mouse")],
+        ... )
+        >>> result = app_thet(aln2)
+        >>> print(result.to_rich_dict())
+        {'type': 'cogent3.app.result.model_result', ...
+
+        Specify the upper and lower bounds for certain branch length and rate exchangeability
+        parameter.
+
+        >>> app_alt_params = get_app(
+        ...     "model",
+        ...     "HKY85",
+        ...     tree=tree,
+        ...     param_rules=[
+        ...         {"par_name": "length", "edge": "Human", "upper": 100, "lower": 1e-2},
+        ...         {"par_name": "kappa", "upper": 200, "lower": 1e-6},
+        ...     ],
+        ... )
+        >>> result = app_alt_params(aln2)
+        >>> print(result.lf.get_param_rules()) # doctest: +NORMALIZE_WHITESPACE
+        [{'par_name': 'kappa', ... 'lower': 1e-06, 'upper': 200}, ...
+        {'par_name': 'length', 'edge': 'Human', ... 'lower': 0.01, 'upper': 100}, ...
+
+        Specify the settings in optimiser.
+
+        >>> app_alt_opt = get_app(
+        ...     "model", "HKY85", tree=tree, opt_args=dict(max_restarts=10, tolerance=1e-8)
```

Comment:
change to `max_restarts=5` and add `max_evaluations=1_000_000, limit_action="ignore"`. Include a description of what those two mean in the text.